docs/source/daml/intro/daml/daml-intro-7/daml/Test/Intro/Asset.daml

```diff
@@ -1,74 +1,77 @@
 -- Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 -- SPDX-License-Identifier: Apache-2.0
-
+{-# LANGUAGE ApplicativeDo #-}
 
 module Test.Intro.Asset where
 
+import Daml.Script
+
 import Intro.Asset
 import Intro.Asset.Role
 
 import Test.Intro.Asset.Role
 
-test_asset = scenario do
+test_asset = do
   ((alice, bob, bank, aha, ahb), assetCid) <- test_issuance
 
   tp <- submit alice do
-    exercise assetCid ProposeTransfer with
+    exerciseCmd assetCid ProposeTransfer with
       newOwner = bob
 
   assetCid <- submit bob do
-    exercise ahb Accept_Transfer with
+    exerciseCmd ahb Accept_Transfer with
       transferProposalCid = tp
 
-  submit bank do
-    asset <- fetch assetCid
-    assert (asset == Asset with
+  asset <- submit bank do
+    createAndExerciseCmd (Helper bank) (FetchAsset assetCid)
+  assert (asset == Asset with
       issuer = bank
       owner = bob
       symbol = "USD"
       quantity = 100.0
       observers = []
         )
 
-test_split = scenario do
+test_split = do
   (setupResult@(alice, bob, bank, _, _), assetCid) <- test_issuance
 
   submitMustFail alice do
-    exercise assetCid Split with
+    exerciseCmd assetCid Split with
       splitQuantity = -30.0
 
   submitMustFail alice do
-    exercise assetCid Split with
+    exerciseCmd assetCid Split with
       splitQuantity = 0.0
 
   submitMustFail alice do
-    exercise assetCid Split with
+    exerciseCmd assetCid Split with
       splitQuantity = 101.0
 
   splitResult <- submit alice do
-    exercise assetCid Split with
+    exerciseCmd assetCid Split with
       splitQuantity = 30.0
 
-  submit bank do
-    splitAsset <- fetch splitResult.splitAsset
-    remainder <- fetch splitResult.remainder
+  (splitAsset, remainder) <- submit bank do
```

Comment:
I think this is not a pattern to teach. Much better to use query.

Reply:
I don’t think you can dismiss this that easily. `query` doesn’t handle divulgence, you have to filter on the client side and it doesn’t allow you to do everything in a single transaction. The latter is not super important for `fetch` but it’s crucial for `fetchByKey` so I’m very hesitant to start telling people to give up on atomicity.

Reply:
Replaced by `queryContractId`